Okta/Keychain.swift

```diff
@@ -0,0 +1,239 @@
+//
```

Comment:
Keeping this as a dependency allows us to consume it willingly, without needing to generate a `THIRD_PARTY_NOTICES` file. If we decided to embed this file, we need:

1. `THIRD_PARTY_NOTICES` file. See [okta-ios-jwt/THIRD_PARTY_NOTICES](https://github.com/okta/okta-ios-jwt/blob/master/THIRD_PARTY_NOTICES) for an example
2. Tests to assert this functionality (can probably copy of what is inside of the consumed lib)

Reply:
Yes, let's use dependencies as much as possible

Reply:
Actually keychain wrapper functionality is pretty simple. It is just one small file. Problem with dependency we had is that it doesn't update since February, and doesn't support Swift 4. In `okta-ios-jwt` repo we use our own `Keychain.swift` (https://github.com/okta/okta-ios-jwt/blob/master/OktaJWT/Keychain.swift). If it is problematic to embed old dependency, let's use Keychain file from `okta-ios-jwt`? Would it be ok?

Reply:
I do agree that we should probably move this logic into the SDK itself. The API to interact with the keychain is pretty straightforward, but will require some additional work on our end as the Keychain util in [`okta-ios-jwt`](https://github.com/okta/okta-ios-jwt/blob/master/OktaJWT/Keychain.swift) doesn't have complete parity with the other dependency.

I see a couple of options:

1. With this PR, stop using `Vinculum` in favor of our own `Keychain` util (can copy over `okta-ios-jwt`'s implementation as a starter). While our test coverage today is OK, we should add additional unit tests to assert this functionality.
2. Use the existing file from `Vinculum` (as currently proposed with this PR), and add a `THIRD_PARTY_NOTICES` file. This should be temporary workaround, as our end goal will be to use our own Keychain util in the future.
3. Wait for the maintainer of the 3rd party library we are using (me) to push a Swift 4 update. I can probably do this over the weekend.

My preference is to do the first option, but understand that this is going to be a bit more work. @robertjd - what do you think?

Reply:
I am also in favor of #1, that seems like the best long-term solution. This lib doesn't need to be dependent on okta-ios-jwt for Keychain stuff.

Reply:
Added own Keychain implementation based on one from JWT